File: LeetCode/answers/src/DailyQuestions/DividePlayersIntoTeamsOfEqualSkill.java
package DailyQuestions;

public class DividePlayersIntoTeamsOfEqualSkill {

    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        long value = skill[0] + skill[skill.length-1];
        long sum = 0;

        for(int index = 0; index < skill.length/2; index++) {
            if(value != skill[index] + skill[skill.length - 1 - index]) {
                return -1;
            }
            sum += skill[index] * skill[skill.length - 1 - index];
        }

        return sum;
    }
}

File: LeetCode/answers/src/DailyQuestions/DividePlayersIntoTeamsOfEqualSkill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        value = skill[0] + skill[-1]
        answer = 0
        while len(skill) != 0 :
            a = skill.pop(0)
            b = skill.pop()
            if a + b != value :
                return -1
            answer += a * b
        return answer
